src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -58,6 +58,12 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
 
             services.TryAddScoped<IVariantFeatureManagerSnapshot>(sp => sp.GetRequiredService<FeatureManagerSnapshot>());
 
+            services.AddSingleton<PercentageFilter>();
```

Comment:
Probably a good idea to call our `AddFeatureFilter` overload. It only adds the filter if it hasn't already been added.

Reply:
Oh, sorry. This is my fault. I found that `services.AddSingleton<PercentageFilter>();` is completly wrong. Because the feature filter will be injected as IFeatureFilterMetadata.

Reply:
Test didn't catch that?

Reply:
All previous testcases called AddFeatureFilter() by themselves. I have not written any new testcases.

Reply:
Sounds like it would be worth to have a test that would've caught the issue.